core/src/test/scala/integration/kafka/api/AuthorizerIntegrationTest.scala

```diff
@@ -0,0 +1,360 @@
+
+      for ((resource, acls) <- RequestKeysToAcls(key))
+        addAndVerifyAcls(acls, resource)
+
+      sendRequestAndVerifyResponseErrorCode(socket, key, requestBytes, ErrorMapping.NoError)
+    }
+  }
+
+    @Test
+    def testProduceNeedsAuthorization() {
+    addAndVerifyAcls(Set(new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Describe)), topicResource)
+      try {
+        sendRecords(numRecords, tp)
+        Assert.fail("should have thrown exception")
+      } catch {
+        case e: ApiException => Assert.assertEquals(Errors.AUTHORIZATION_FAILED.exception().getMessage, e.getMessage)
+      }
+    }
+
+    @Test
+    def testOnlyWritePermissionAllowsWritingToProducer() {
+      addAndVerifyAcls(Set(new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Write)), topicResource)
+      try {
+        sendRecords(numRecords, tp)
+      } catch {
+        case e: Exception => Assert.fail("Should not fail")
+      }
+    }
+
+    @Test
+    def testCreatePermissionNeededForWritingToNonExistentTopic() {
+      val newTopic = "newTopic"
+      val topicPartition = new TopicPartition(newTopic, 0)
+      addAndVerifyAcls(Set(new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Write)), new Resource(Topic, newTopic))
+      try {
+        sendRecords(numRecords, topicPartition)
+        Assert.fail("should have thrown exception")
+      } catch {
+        case e: TimeoutException =>
+        //TODO Need to update the producer so it actually throws the server side of exception.
+        case e: Exception => Assert.fail("Only timeout exception should be thrown.")
+      }
+
+      addAndVerifyAcls(Set(new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Create)), Resource.ClusterResource)
+      try {
+        sendRecords(numRecords, topicPartition)
+      } catch {
+        case e: Exception => Assert.fail("No exception should be thrown.")
+      }
+    }
+
+    @Test
+    def testConsumerNeedsAuthorization() {
+      addAndVerifyAcls(Set(new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Write)), topicResource)
+      //TODO: Ideally we would want to test that when consumerGroup permission is not present we still get an AuthorizationException
+      //but the consumer fetcher currently waits forever for the consumer metadata to become available.
+      addAndVerifyAcls(Set(new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Read)), groupResource)
+      sendRecords(1, tp)
+      try {
+        this.consumers.head.assign(List(tp).asJava)
+        consumeRecords(this.consumers.head)
+        Assert.fail("should have thrown exception")
+      } catch {
+        case e: ApiException => Assert.assertEquals("Not authorized to read from topic-0", e.getMessage)
+      }
+    }
+
+    @Test
+    def testAllowingReadOnTopicAndGroupAllowsReading() {
+      addAndVerifyAcls(Set(new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Write)), topicResource)
```

Comment:
I originally thought I need to have greater than 1 identity but as I wrote tests I did not see the need. Using SSL certificate will probably test the SSL auth part in addition to authorizer but will not change the validity of these tests in anyway. Let me know if you think otherwise.